### PR TITLE
Feature/root pom 15 multi arch builds

### LINKED ISSUE #15: Docker: Multi-arch build
Add support for multi-arch builds.

For example:
 ```shell
mvn -Ddocker ... -Ddocker.platforms=linux/amd64,linux/arm64
```

### PR DESCRIPTION
**Pull request by wajda**
Fixes #15 

- Introduce a new property `docker.platforms` that enable multi-platform builds. Change the `maven-docker-plugin` config accordingly.
- Add renamed aliases `docker.` for the existing properties `dockerfile.*`, and deprecate the later ones for brevity and consistency reasons. Backward compatibility with the existing client POMs is preserved.
- Add a forked copy of the docker plugin to fix an auth issue on `docker.io` (see fabric8io/docker-maven-plugin#1688). This change is likely temporary, until the PR is merged into the main repo reasonably quickly.

### CODE REVIEW

**Review by cerveada — DISMISSED** [review]
LGTM, but if it doesn't take too long, it's better to wait for the official release of the plugin.

**Comment by wajda** [comment]
> LGTM, but if it doesn't take too long, it's better to wait for the official release of the plugin.

The PR to the "maven-docker-plugin" has been merged. According to the maintainer, a new version is expected to be released by Monday, July 31.
https://github.com/fabric8io/docker-maven-plugin/pull/1694#issuecomment-1648039371

**Comment by wajda** [comment]
The fixed version of the Maven plugin has been published, so I revert my last temporary commit, update the version of the plugin, and merging the PR.

**Pull request merged**
